internal/temporalcommonv1/payload_json_test.go

```diff
@@ -102,6 +102,30 @@ var tests = []struct {
 		},
 		Data: []byte(`{"greeting":{"name":{}}}`),
 	},
+}, {
+	name:          "empty payloads",
+	longformJSON:  `{}`,
+	shorthandJSON: `null`,
```

Comment:
Hrmm, arguably this could be `[]`

Reply:
An empty array seems "correct" as it is an empty array, but we should support parsing either